client/signup/steps/domains/index.jsx

```diff
@@ -708,14 +713,17 @@ export class RenderDomainsStep extends Component {
 			)
 		);
 
-		const productToRemove = this.props.cart.products.find(
-			( product ) => product.product_slug === this.props.multiDomainDefaultPlan.product_slug
+		const productToRemove = cart.products.find(
+			( product ) => product.product_slug === multiDomainDefaultPlan.product_slug
 		);
-		const uuidToRemove = productToRemove.uuid;
 
-		this.props.shoppingCartManager.removeProductFromCart( uuidToRemove ).then( () => {
-			this.props.goToNextStep();
-		} );
+		if ( productToRemove && productToRemove.uuid ) {
+			shoppingCartManager.removeProductFromCart( productToRemove.uuid ).then( () => {
+				goToNextStep();
+			} );
+		} else {
+			goToNextStep();
+		}
```

Comment:
The e2e tests failed in https://github.com/Automattic/wp-calypso/pull/84070 because I forgot to conditionally remove the "hidden plan". So, if the user selected a free wordpress.com plan and clicked "Continue" it got stuck waiting for the non-existent plan to be removed.